prov/net/src/xnet_progress.c

```diff
@@ -1479,10 +1479,17 @@ int xnet_init_progress(struct xnet_progress *progress, struct fi_info *info)
 
 void xnet_close_progress(struct xnet_progress *progress)
 {
+	struct xnet_event *event;
+	struct slist_entry *item;
+
 	assert(dlist_empty(&progress->unexp_msg_list));
 	assert(dlist_empty(&progress->unexp_tag_list));
 	assert(dlist_empty(&progress->saved_tag_list));
-	assert(slist_empty(&progress->event_list));
+	while (!slist_empty(&progress->event_list)) {
+		item = slist_remove_head(&progress->event_list);
+		event = container_of(item, struct xnet_event, list_entry);
+		free(event);
+	}
```

Comment:
It might work to clean up the list for this test, but I'm concerned about the problem of having events that reference destroyed endpoints.  That could show up for other apps that have other active endpoints and don't close the progress engine.  They would read these events, try to process them, and likely access freed memory.

Maybe we need some sort of cleanup in the rdm ep close path for this instead.

Reply:
Thanks! I have removed the last commit, as you suggested. 
I will address that particular issue in a new PR.